scm-core/src/main/java/sonia/scm/repository/RepositoryContentInitializer.java

```diff
@@ -57,10 +69,18 @@
 
     /**
      * create new file which will be included in initial repository commit
+     *
      * @param path path of new file
      * @return
      */
     CreateFile create(String path);
+
+    /**
+     * @return creation context of repository which is going to be initialized
+     */
+    default Map<String, JsonNode> getCreationContext() {
+      return Collections.emptyMap();
+    }
```

Comment:
I don't like this api. It should be as easy as possible for bob (the plugin author).  I think an api like the following would be much easier to use:

```suggestion

    /**
     * Find the the context object with the given key and unmarshalls it to the given type.
     *
     * @param key key of the context object
     * @param type type of the context object
     *
     * @return context object or empty optional
     * @since 2.5.0
     */
    default <T> Optional<T> fromContext(String key, Class<T> type) {
      return Optional.empty();
    }
```